### ISSUE TITLE
parse a unit type from a string

### ISSUE THREAD

**Issue by happysalada** [iliekturtles/uom#391]
Is there a way to parse a unit type from a string ?
I need to send some values with their unit types to databases, so I was thinking of converting the unit type to a string to store it. I don't see any way of deserializing it from a string though, did I miss something ?

**Comment by iliekturtles** [comment]
Quantity implements [`FromStr`](https://github.com/iliekturtles/uom/blob/6de9f7b3bfe5600b52c3e4b80f52c9b58c9347cf/src/quantity.rs#L428). Is this what you're looking for or could you provide a bit more detail?

```rust
use uom::si::f32::Mass;

let l = "1 km".parse::<Mass>().unwrap();
```

**Comment by happysalada** [comment]
Right, i was looking to use the unit types only, not directly a quantity. I was thinking of just separating the value and the unit in a struct.
I realized that this is not how this library was designed to be used though. Thank you for yout reply !

**Comment by iliekturtles** [comment]
Yes, `uom` isn't designed to work at the unit level currently. May this issue be closed, or did you still have an open issue looking to be resolved?

**Comment by happysalada** [comment]
Thanks again !
